tests/Unit/NumericalAlgorithms/RootFinding/Test_QuadraticEquation.cpp

```diff
@@ -42,6 +44,50 @@
 #endif
 }
 
+namespace {
+  template <typename T>
+  void test_smallest_root_greater_than_value_within_roundoff(
+      const T& used_for_size) noexcept {
+    const auto a = make_with_value<T>(used_for_size, 2.0);
+    const auto b = make_with_value<T>(used_for_size, -11.0);
+    const auto c = make_with_value<T>(used_for_size, 5.0);
+    const auto expected_root_1 = make_with_value<T>(used_for_size, 0.5);
+    const auto expected_root_2 = make_with_value<T>(used_for_size, 5.0);
+
+    auto root = smallest_root_greater_than_value_within_roundoff(a, b, c, 0.3);
+    CHECK_ITERABLE_APPROX(expected_root_1, root);
+    root = smallest_root_greater_than_value_within_roundoff(a, b, c, 0.5);
```

Comment:
What about a check with a value of 0.5 + epsilon, to check the "within roundoff" condition?